From bbb314351e7224b2f7e8ffe2e34d17c76edc124b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?R=C3=A9mi=20Bernon?= <rbernon@codeweavers.com>
Date: Thu, 3 Sep 2020 15:36:44 +0200
Subject: [PATCH 07/11] gdi32/tests: Add todo_wine for cmmi10 font and alike.

This test is failing for several system fonts, that fall in some
unsupported corner case. The case is commented in freetype.c:

    TM.tmLastChar = pOS2->usLastCharIndex;   /* Should be min(cmap_last, os2_last) */
---
 dlls/gdi32/tests/font.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/dlls/gdi32/tests/font.c b/dlls/gdi32/tests/font.c
index 8b0c7b3dbcc..afe1f82fd09 100644
--- a/dlls/gdi32/tests/font.c
+++ b/dlls/gdi32/tests/font.c
@@ -3879,6 +3879,8 @@ static void test_text_metrics(const LOGFONTA *lf, const NEWTEXTMETRICA *ntm)
                tmA.tmFirstChar == expect_first_A + 1 /* win9x */,
                "A: tmFirstChar for %s got %02x expected %02x\n", font_name, tmA.tmFirstChar, expect_first_A);
         if (pGdiGetCodePage == NULL || ! IsDBCSLeadByteEx(pGdiGetCodePage(hdc), tmA.tmLastChar))
+            todo_wine_if( cmap_type != cmap_ms_symbol && (cmap_first < 0xf000 || cmap_first >= 0xf100) &&
+                          os2_last_char < 0xff && os2_last_char < expect_last_W )
             ok(tmA.tmLastChar == expect_last_A ||
                tmA.tmLastChar == 0xff /* win9x */,
                "A: tmLastChar for %s got %02x expected %02x\n", font_name, tmA.tmLastChar, expect_last_A);
-- 
2.29.2

